### PR TITLE
Loadgen cleanup

### LINKED ISSUE #4063: Clean up Soroban Loadgen
- [x] Cleanup `GeneratedLoadConfig`
- [x] Better status logs for loadgen
- [x] Metrics for Soroban loads
- [x] Soroban state reset function
- [ ] Discrete distribution for soroban loadgen parameters instead of random

### PR DESCRIPTION
**Pull request by SirTyson**
# Description

Mostly resolves #4063.

Additionally, `SOROBAN_INVOKE` mode now takes the total amount of IO kilobytes as a parameter instead of kilobytes per data entry, making it possible to write tests that respect network config bounds while still having variance in the number of data entries written.

# Checklist
- [x] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [x] Rebased on top of master (no merge commits)
- [x] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [x] Compiles
- [x] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/simulation/LoadGenerator.cpp [hunk 1]
```diff
@@ -1208,33 +1303,31 @@ LoadGenerator::invokeSorobanLoadTransaction(uint32_t ledgerNum,
...
+        kiloBytesPerEntry =
+            (totalWriteBytes - mContactOverheadBytes) / numEntries / 1024;
 
-    maxKiloBytesPerEntry =
-        std::min(maxKiloBytesPerEntry, cfg.kiloBytesPerDataEntryHigh);
-    uint32_t minKiloBytesPerEntry =
-        std::min(maxKiloBytesPerEntry, cfg.kiloBytesPerDataEntryLow);
-    auto kiloBytesPerEntry =
-        rand_uniform<uint32_t>(minKiloBytesPerEntry, maxKiloBytesPerEntry);
+        // If numEntries > 0, we can't write a 0 byte entry
```
Comment (dmkozh) [inline src/simulation/LoadGenerator.cpp:1325]: Hmm, we actually can, can't we? It won't be 0 bytes, probably about 8 bytes to write an empty Bytes ScVal.

File: src/simulation/LoadGenerator.cpp [hunk 2]
```diff
@@ -1250,30 +1343,26 @@ LoadGenerator::invokeSorobanLoadTransaction(uint32_t ledgerNum,
...
     // baseInstructionCount is a very rough estimate and may be a significant
     // underestimation based on the IO load used, so use max instructions
     resources.instructions = networkCfg.mTxMaxInstructions;
-    resources.readBytes = readBytesUpperBound;
-    resources.writeBytes = writeBytes;
+
+    // We don't have a good way of knowing how many bytes we will need to read
+    // since the previous invocation writes a random number of bytes, so use
+    // upper bound
+    resources.readBytes = invokeCfg.ioKiloBytesHigh * 1024;
```
Comment (dmkozh) [inline src/simulation/LoadGenerator.cpp:1353]: Why did you change this back to the max value?

File: src/simulation/LoadGenerator.cpp (hunk 1 above)
Comment (SirTyson) [reply src/simulation/LoadGenerator.cpp:1325]: Sorry, I didn't know this was valid behavior. Currently the loadgen contract panics if `size_kilo_bytes == 0 && num_write_entries != 0`. We can fix this later, but personally I don't think an env change and wasm recompile is worth fixing this right now.

File: src/simulation/LoadGenerator.cpp (hunk 2 above)
Comment (SirTyson) [reply src/simulation/LoadGenerator.cpp:1353]: This is the same value that we had before. Previously this was 
```
readBytes = (cfg.kiloBytesPerDataEntryHigh * 1024 + leOverhead) * numEntries + mContactOverheadBytes;
```
Since the config parameter now dictates total TX io size instead of the size of individual data entries, we can simplify it. Note that this is the LoadGen config max, not the network settings max. Unfortunately we have a lot of different config types floating around so it's a bit confusing.

File: src/simulation/LoadGenerator.cpp (hunk 1 above)
Comment (dmkozh) [reply src/simulation/LoadGenerator.cpp:1325]: Ah, ok. Empty bytes sequence is a valid value for a byte sequence, but I agree there is no need to update Wasm due to that. Also, it doesn't provide any significant value for coverage.

File: src/simulation/LoadGenerator.cpp (hunk 2 above)
Comment (dmkozh) [reply src/simulation/LoadGenerator.cpp:1353]: Ah, right, I've got confused by this a bit.

**Comment by dmkozh** [comment]
r+ 113f5d9

**Comment by marta-lokhova** [comment]
r+ 8b825c06c95240c58992ccbf5ff2ed6d02b30df2
